core/dbt/task/debug.py

```diff
@@ -129,6 +130,16 @@ def run(self):
             print(message)
             print('')
 
+        if self.any_failure:
+            sys.exit(ExitCodes.ModelError)
```

Comment:
I don't think we want to short circuit here! Exit codes are handled in the main entrypoint. So we just need to return correct value for `interpret_results` in this task.

https://github.com/fishtown-analytics/dbt/blob/1f927a374c8bd52a12a20d892fed9d59cffd04f4/core/dbt/main.py#L125-L129

Reply:
that's on me for recommending `sys.exit()` here! much better, thanks @kwigley